File: leetcode/896. Monotonic Array/main.ex
defmodule Solution do
  @spec is_monotonic(nums :: [integer]) :: boolean
  def is_monotonic(nums) do
    Enum.chunk_every(nums, 2, 1, :discard)
    |> Enum.map(fn
        [a, b] when a < b -> -1
        [a, b] when a > b -> 1
        _ -> 0
    end)
    |> Enum.uniq()
    |> Enum.sort()
    |> IO.inspect()
    |> (fn list -> list != [-1, 1] and list != [-1, 0, 1] end).()
  end
end

File: leetcode/896. Monotonic Array/main.rs
use std::cmp::Ordering;

impl Solution {
    pub fn is_monotonic(nums: Vec<i32>) -> bool {
        if nums.len() == 1 {
            return true;
        }
        // save current order of array
        let mut ord = None;
        for i in 0..(nums.len() - 1) {
            let cur = nums[i].cmp(&nums[i+1]);
            match (ord, cur) {
                // If the order reverses, it is not monotonically inc/dec
                (Some(Ordering::Less), Ordering::Greater) | (Some(Ordering::Greater), Ordering::Less) => return false,
                // Only update the order once when we know it (and it's not Ordering::Equal)
                (None, Ordering::Greater) | (None, Ordering::Less) => {
                    ord = Some(cur);
                }
                _ => {}
            }
        }
        return true;
    }
}